### ISSUE TITLE
Improve the URL descriptions

### ISSUE THREAD

**Issue by djzwerg** [backdrop-contrib/matomo#27]
Because the trailing slash is importing to make the tracking work a hint should be added to the desciption.

**Comment by indigoxela** [comment]
> Because the trailing slash is importing

You meant "important", right? :wink: 

Problem: the leading slash is only an issue, if Matomo runs on the same Vhost. Many people either have a dedicated Vhost, or run Matomo as a service (external), so they provide full urls. Even if it's on the same Vhost, people can still provide the full url.

I'm not sure if it's really helpful to expand the (already a little long) help text even more - with info for a minority of users.
More examples in the README, or maybe a Wiki page with way more detailed instructions could be more helpful. Especially for those not familiar with things like relative or absolute urls.

**Comment by indigoxela** [comment]
Another concern: actually the plan is to consolidate these two input fields. It makes no sense to set the same thing twice, and it also makes no sense to set an http url, when a https url would be available. But I'm currently too busy for that.

@djzwerg What do you thing about a single input field? And better instructions either in the README or in the Wiki?

**Comment by djzwerg** [comment]
Yes only one input field should be the right way. Because of law (GDPR) and security we should force HTTPS instead of HTTP. But that should be another issue imho.

I think transfer contents of the README file to the Wiki will be the 'right Backdrop way' (maybe with links to Matomo docs).

**Comment by olafgrabienski** [comment]
>> Because the **trailing slash** ...

> Problem: the **leading slash** ...

@indigoxela This isn't about the *leading* but the *trailing* slash. If it isn't present, it causes issues even for full URLs.

edit: I guess the confusion comes from the [PR](https://github.com/backdrop-contrib/matomo/pull/28/files) which mentions a "leading" slash. Should be "trailing" also there.

**Comment by olafgrabienski** [comment]
Is it possible to check if the *trailing* slash is present, and if not, to let the module add it?

**Comment by olafgrabienski** [comment]
In any case, here's my suggestion for the description text:

> The URL to your Matomo base directory, including the trailing slash. Example: `http://www.example.com/matomo/`

It's a bit less noisy in comparison to the PR. Also, with the example displayed as `code` and without the period sign (full stop) at the end, the trailing slash is slightly better visible.

**Comment by indigoxela** [comment]
The actual plan:

- Reduce the url input fields to only one.
- Catch all leading/trailing slash input problems on form submit (fix it automatically in code)

But not today... stay tuned. It needs a little overhaul of the form submission handling and an update hook for the input reduction.

**Comment by indigoxela** [comment]
There hasn't been much activity here, and the discussion seems a bit confusing. As the PR hasn't been updated, either, I'm closing this in favor of #48
